### PR TITLE
Updates test coverage

### PR DESCRIPTION
**Pull request by ebenolson**
To add some coverage of the update rules, I wrote a test comparing their results to values calculated using the Torch optim package. 

In the process, I found a discrepancy with the ADAM updates, after looking at the paper I think the Torch version is correct but I'd appreciate a second opinion.

### CODE REVIEW

**Comment by ebenolson** [comment]
:tired_face:  not sure how this happened, but I apparently deleted `total_norm_constraint()` by accident then reimplemented something similar...  will need to rework this tomorrow

**Comment by benanne** [comment]
Good catch on the Adam implementation, I didn't spot that when it was merged. I guess those numerical tests are going to prove useful :)

**Comment by benanne** [comment]
It would be useful to add a comment describing briefly how those values were obtained using Torch. That way if we decide to add more algorithms later we can update the tests more easily as well.

**Comment by ebenolson** [comment]
Ok, I fixed the accidental deletion and added a link to a gist of the torch script I used.

**Comment by skaae** [comment]
The update tests are nice. Good catch with the adam bug I don't know where i got the squared idea from :) I probably messed up in the dense subscript/superscript notation.

**Comment by benanne** [comment]
Looks good! I'm okay with merging if there are no further comments.

File: lasagne/tests/test_updates.py [hunk 1]
```diff
@@ -64,7 +139,7 @@ def test_norm_constraint_dim6_raises():
     assert "Unsupported tensor dimensionality" in str(excinfo.value)
 
 
-def total_norm_constraint():
+def test_total_norm_constraint():
```
Comment (f0k) [inline lasagne/tests/test_updates.py]: Ooops, didn't see that when reviewing.

**Comment by f0k** [comment]
Nice work! Also an interesting toy problem to demonstrate how advanced methods are independent to gradient magnitude.

**Comment by benanne** [comment]
Alright then, merging!

**Pull request merged**